File: Concept_Programs/Single.cpp
#include<iostream>
using namespace std;

class Base
{
    public:
        int A,B;
        Base()
        {   
            cout<<"Inside Base constructor\n";
            A=11;
            B=21;
        }
        ~Base()
        {   cout<<"Inside Base destructor\n";   }
        void Fun()
        {   cout<<"Inside Base fun\n";  }
};
class Derived : public Base
{
    public:
        int X,Y;
        Derived()
        {   
            cout<<"Inside Derived constructor\n";
            X=51;
            Y=101;
        }
        ~Derived()
        {   cout<<"Inside Derived destructor\n";    }
        void Gun()
        {   cout<<"Inside Derived gun\n";   }
};

int main()
{
    Derived dobj;
    cout<<"Size of object is: "<<sizeof(dobj)<<"\n";
    
    cout<<dobj.A<<"\n";
    cout<<dobj.B<<"\n";
    cout<<dobj.X<<"\n";
    cout<<dobj.Y<<"\n";

    dobj.Fun();
    dobj.Gun();
    
    return 0;
}

File: Concept_Programs/Single.java

class Base
{
    public int A,B;
    public Base()
    {
        System.out.println("Base constructor");
    }
    public void Fun()
    {
        System.out.println("Inside Base fun");
    }
}

class Derived extends Base
{
    public int X,Y;
    public Derived()
    {
        System.out.println("Derived constructor");
    }
    public void Gun()
    {
        System.out.println("Inside Derived gun");
    }
}

class Single
{
    public static void main(String Arg[])
    {
        Derived dobj=new Derived();
        dobj.Fun();
        dobj.Gun();
    }
}